src/main/java/io/jenkins/plugins/analysis/core/steps/PublishIssuesStep.java

```diff
@@ -572,7 +576,13 @@ protected ResultAction run() throws IOException, InterruptedException, IllegalSt
             IssuesPublisher publisher = new IssuesPublisher(getRun(), report, healthDescriptor, qualityGate,
                     name, referenceJobName, ignoreQualityGate, ignoreFailedBuilds,
                     getCharset(sourceCodeEncoding), getLogger(report));
-            return publisher.attachAction();
+            ResultAction resultAction = publisher.attachAction();
+            QualityGateStatus status = resultAction.getResult().getQualityGateStatus();
+            if (status == QualityGateStatus.WARNING) {
+                WarningAction action = new WarningAction("Some quality gates have been missed: overall result is " + status);
```

Comment:
We should have a subclass of `WarningAction` here (or your action should be an interface?). Otherwise I can't provide a localized message. So `WarningAction.getMessage()` can be overwritten with a localized string.

Reply:
That's a good point. Note that as of right now, the message for `WarningAction` is not actually displayed/used anywhere. That said, I imagine it would be used in contexts such as visualizations, where a localized message would probably make sense rather than in something like builds logs where it would make less sense. Let me think about whether to make `WarningAction` itself take optionally take a `Localizable` or if plugins should subclass the action. In some cases `WarningAction` will be created from user-generated text, so those cases wouldn't be localizable, but I'd guess that plugins would like to be able to provide localizable messages when they use the actions.

Reply:
I'm not sure if using a `Localizable` property in a serializable action is a good thing (will be part of build.xml).

Reply:
The [`Localizable` class and its members are `Serializable`](https://github.com/kohsuke/localizer/blob/a3e50a68020e94505899d5c754f61ead66ae53ab/lib/src/main/java/org/jvnet/localizer/Localizable.java#L39), but yeah perhaps it's not the best idea to actually serialize it.

Reply:
For now, I made `WarningAction` non-final, so it could be extended here to support localization, although there are still no uses of the message so I'm not sure what contexts it would be displayed in and whether localization would make sense in those contexts.